<div align="center">

# 📊 S13 — NumPy aplicado al análisis de portafolios
### Lenguajes de Programación · Ingeniería Financiera

*De la lista al array — y del array a una recomendación de inversión*

</div>

---

En el reto del portafolio (S11-S12) escribiste tus propias funciones. Hoy conoces NumPy — y lo vas a usar no solo para calcular, sino para **analizar y justificar decisiones**, como lo haría un analista de verdad.

| Bloque | Tema |
|:---:|---|
| 00 | ¿Qué es una librería? |
| 01 | Qué es un array — y cuántas dimensiones puede tener |
| 02 | Creación de arrays |
| 03 | Operaciones básicas |
| 04 | Analiza un activo: rendimiento y riesgo en el tiempo |
| 05 | Compara los 4 activos |
| 06 | Construye un portafolio ponderado |
| 07 | 🎯 Reto individual |

## 00 · ¿Qué es una librería?

> **Nota** — Objetivo: entender qué es una librería antes de usar una.

Una librería es un paquete de funciones que otra persona ya escribió, probó y publicó para que cualquiera las use — como las tuyas, pero construidas para casos generales. **NumPy** es una librería con funciones ya hechas para trabajar con muchos números a la vez.

**¿Cuál es el objetivo de NumPy?** Darle a Python la capacidad de hacer cálculos numéricos rápido sobre conjuntos grandes de datos.

In [ ]:
import numpy as np

print("NumPy", np.__version__, "listo para usar.")

> 💡 **Idea clave** — `import` le dice a Python que traiga la librería. `numpy` es su nombre. `as np` le da un apodo corto — es la convención que usa toda la comunidad.

---
## 01 · Qué es un array — y cuántas dimensiones puede tener

> **Nota** — Objetivo: entender qué es un array, y que puede tener una o más dimensiones.

Un array es una colección de números, todos del mismo tipo, donde cada uno tiene una posición fija — su **índice**.

In [2]:
import numpy as np

precios = np.array([195, 385, 185, 165])   # AAPL, MSFT, AMZN, GOOGL

print(precios)
print("shape:", precios.shape)
print("dtype:", precios.dtype)

[195 385 185 165]
shape: (4,)
dtype: int64


Esta fila tiene **una sola dimensión** — un **eje (axis)**. Pero un array puede tener dos ejes: una tabla, con filas y columnas. Por ejemplo, el precio de 4 activos durante 5 días:

| | Día 1 | Día 2 | Día 3 | Día 4 | Día 5 |
|---|---|---|---|---|---|
| **AAPL** | 192 | 194 | 193 | 196 | 195 |
| **MSFT** | 390 | 388 | 386 | 384 | 385 |

In [3]:
precios_5_dias = np.array([
    [192, 194, 193, 196, 195],   # AAPL
    [390, 388, 386, 384, 385],   # MSFT
])

print(precios_5_dias)
print("shape:", precios_5_dias.shape, "  # (2 activos, 5 días)")

[[192 194 193 196 195]
 [390 388 386 384 385]]
shape: (2, 5)   # (2 activos, 5 días)


> 💡 **Idea clave** — Más adelante en esta clase, cada activo va a tener justo esta forma: **una fila de rendimientos a través de varios días**. Guarda esta idea — es la base de todo lo que sigue.

---
## 02 · Creación de arrays

> **Nota** — Objetivo: crear un array a partir de una lista.

**✏️ Completa el código**

In [5]:
cantidades = np.array([10,8,12,6])   # 10, 8, 12, 6
rendimientos = np.array([8.33,-3.75,8.82,10.00]) # 8.33, -3.75, 8.82, 10.00

print(cantidades, cantidades.dtype)
print(rendimientos, rendimientos.dtype)

[10  8 12  6] int64
[ 8.33 -3.75  8.82 10.  ] float64


<details>
<summary>💡 Ver respuesta</summary>

```python
cantidades = np.array([10, 8, 12, 6])
rendimientos = np.array([8.33, -3.75, 8.82, 10.00])
```
</details>

> ⚠️ **Error frecuente** — `np.array(1, 2, 3, 4)` falla: los números deben ir dentro de **una sola lista**. Pruébalo (esto debe fallar):

In [8]:
np.array([1, 2, 3, 4])

array([1, 2, 3, 4])

---
## 03 · Operaciones básicas

> **Nota** — Objetivo: operar el array completo, sin ciclos.

Los operadores aritméticos se aplican **elemento por elemento**.

In [9]:
precios_actuales = np.array([195, 385, 185, 165])
precios_compra = np.array([180, 400, 170, 150])

valor_invertido = cantidades * precios_compra
valor_actual = cantidades * precios_actuales

print("Valor actual:", valor_actual)

Valor actual: [1950 3080 2220  990]


**✏️ Completa el código** — calcula `ganancia` y `en_perdida` (verdadero/falso).

In [10]:
ganancia = valor_actual - valor_invertido
en_perdida = ganancia < 0

print("Ganancia:", ganancia)
print("En pérdida:", en_perdida)

Ganancia: [ 150 -120  180   90]
En pérdida: [False  True False False]


<details>
<summary>💡 Ver respuesta</summary>

```python
ganancia = valor_actual - valor_invertido
en_perdida = ganancia < 0
```
</details>

> 💡 **Idea clave** — Ese array de `True`/`False` es exactamente lo que se usa por dentro para **filtrar** una tabla en pandas (S14).

---
## 04 · Analiza un activo: rendimiento y riesgo en el tiempo

> **Nota** — Objetivo: usar `mean()` y `std()` para describir el comportamiento de un activo a través de varios días — no solo calcularlos.

A partir de aquí trabajas con algo distinto: no es "un número por activo", sino **la serie de rendimientos diarios de AAPL durante 8 días** de negociación.

In [11]:
AAPL = np.array([0.012, 0.008, -0.004, 0.015, 0.006, -0.002, 0.010, 0.003])

print("Número de días:", len(AAPL))
print("Primer día:", AAPL[0])
print("Último día:", AAPL[-1])

Número de días: 8
Primer día: 0.012
Último día: 0.003


**🔎 Pregunta de análisis**

Un compañero mira el promedio de AAPL y afirma: *"Como el rendimiento promedio es positivo, quiere decir que AAPL tuvo un buen día todos los días."*

¿Tiene razón? Usa el propio array `AAPL` (no solo el promedio) para comprobarlo antes de responder.

In [12]:
promedio_aapl = AAPL.mean()
peor_dia = AAPL.min()

print("Promedio:", promedio_aapl)
print("Peor día:", peor_dia)

Promedio: 0.006
Peor día: -0.004


<details>
<summary>💡 Ver una forma de responder</summary>

No tiene razón. El promedio (0.6%) es positivo, pero el peor día fue **-0.4%** — hubo al menos un día con pérdida. El promedio resume la tendencia general, no garantiza que cada día individual haya sido positivo. Para saber eso hay que mirar el array completo, no solo su resumen.
</details>

**✏️ Ahora tú:** calcula el riesgo de AAPL (su desviación estándar).

In [13]:
riesgo_aapl = AAPL.std()
print("Riesgo (std):", riesgo_aapl)

Riesgo (std): 0.006224949798994366


<details>
<summary>💡 Ver respuesta</summary>

```python
riesgo_aapl = AAPL.std()
```
`std()` ≈ 0.0062 (0.62%).
</details>

Ese número mide qué tanto se aleja cada día del promedio, en ambas direcciones — no cuánto ganas o pierdes en un día cualquiera, sino qué tan predecible o errático fue el activo durante esos 8 días. Por eso a la desviación estándar se le trata, en la práctica, como sinónimo de **volatilidad**: dos activos pueden tener el mismo rendimiento promedio y aun así uno ser mucho más riesgoso que el otro, simplemente porque sus días buenos y malos son más extremos.

---
## 05 · Compara los 4 activos

> **Nota** — Objetivo: comparar rendimiento y riesgo de varios activos a la vez, y tomar una posición justificada — no solo calcular cada uno por separado.

Estas son las series de 8 días de tus 4 activos:

In [14]:
AAPL = np.array([0.012, 0.008, -0.004, 0.015, 0.006, -0.002, 0.010, 0.003])
MSFT = np.array([0.004, 0.003, 0.005, 0.002, 0.006, 0.004, 0.003, 0.005])
AMZN = np.array([-0.010, -0.015, 0.002, -0.008, -0.012, 0.005, -0.020, -0.006])
TSLA = np.array([0.030, -0.025, 0.040, 0.015, -0.020, 0.035, -0.010, 0.020])

**✏️ Completa la tabla** — calcula el rendimiento promedio y el riesgo de cada uno.

In [17]:
activos = {"AAPL": AAPL, "MSFT": MSFT, "AMZN": AMZN, "TSLA": TSLA}

for nombre, serie in activos.items():
    promedio = serie.mean()
    riesgo = serie.std()
    print(f"{nombre}: rendimiento {promedio:.4f}  |  riesgo {riesgo:.4f}")

AAPL: rendimiento 0.0060  |  riesgo 0.0062
MSFT: rendimiento 0.0040  |  riesgo 0.0012
AMZN: rendimiento -0.0080  |  riesgo 0.0078
TSLA: rendimiento 0.0106  |  riesgo 0.0239


<details>
<summary>💡 Ver respuesta</summary>

```python
for nombre, serie in activos.items():
    promedio = serie.mean()
    riesgo = serie.std()
    print(f"{nombre}: rendimiento {promedio:.4f}  |  riesgo {riesgo:.4f}")
```

| Activo | Rendimiento promedio | Riesgo (std) |
|---|---|---|
| AAPL | 0.60% | 0.62% |
| MSFT | 0.40% | 0.12% |
| AMZN | -0.80% | 0.78% |
| TSLA | 1.06% | 2.39% |
</details>

**🔎 Preguntas de análisis** — respóndelas en una celda de texto, con la tabla de arriba como evidencia:

1. Si **solo** te importara maximizar el rendimiento promedio, ¿qué activo elegirías? Si **solo** te importara minimizar el riesgo, ¿cuál elegirías? ¿Es el mismo activo en los dos casos? ¿Qué te dice eso?
2. Un cliente te dice: *"Nunca invertiría en algo con tanto riesgo como TSLA."* Otro cliente te dice: *"Nunca invertiría en algo que en promedio pierde dinero, como AMZN."* ¿Estás de acuerdo con ambos? Justifica usando los números de la tabla, no solo tu opinión.
3. Recomienda **un solo activo** para un cliente que prioriza la estabilidad por encima de todo. Justifica tu elección considerando rendimiento *y* riesgo — no solo uno de los dos.

---
## 06 · Construye un portafolio ponderado

> **Nota** — Objetivo: combinar varios activos con distintos pesos, y entender qué le pasa al riesgo cuando diversificas.

Un portafolio real casi nunca es "todo en un solo activo" — se reparte el dinero entre varios, con distintos pesos. Supón esta distribución:

In [18]:
pesos = np.array([0.35, 0.40, 0.10, 0.15])  # AAPL, MSFT, AMZN, TSLA — suman 1.0

rendimiento_portafolio = AAPL*pesos[0] + MSFT*pesos[1] + AMZN*pesos[2] + TSLA*pesos[3]
print(rendimiento_portafolio)

[ 0.0093  -0.00125  0.0068   0.0075   0.0003   0.00665  0.0012   0.00545]


> 💡 **Idea clave** — Una sola operación combinó los 4 activos, día por día, sin ningún `for`. Cada día del portafolio es el promedio ponderado de lo que hicieron los 4 activos ese día.

**✏️ Completa el código** — calcula el rendimiento promedio y el riesgo del portafolio.

In [ ]:
rendimiento_promedio_portafolio = rendimiento_portafolio.mean()
riesgo_portafolio = rendimiento_portafolio.std()

print(f"Rendimiento promedio del portafolio: {rendimiento_promedio_portafolio:.4f}")
print(f"Riesgo del portafolio: {riesgo_portafolio:.4f}")

<details>
<summary>💡 Ver respuesta</summary>

```python
rendimiento_promedio_portafolio = rendimiento_portafolio.mean()
riesgo_portafolio = rendimiento_portafolio.std()
```

Rendimiento promedio ≈ **0.45%**, riesgo ≈ **0.36%**.
</details>

**🔎 Pregunta de análisis**

Compara el riesgo del portafolio (≈0.36%) con el promedio ponderado de los riesgos *individuales* de los 4 activos, calculado abajo:

In [21]:
riesgos_individuales = np.array([AAPL.std(), MSFT.std(), AMZN.std(), TSLA.std()])
riesgo_promedio_simple = (riesgos_individuales * pesos).sum()

# Redefinir riesgo_portafolio para asegurar que esté disponible
riesgo_portafolio = rendimiento_portafolio.std()

print(f"Riesgo si solo promediaras los riesgos individuales: {riesgo_promedio_simple:.4f}")
print(f"Riesgo real del portafolio combinado: {riesgo_portafolio:.4f}")

Riesgo si solo promediaras los riesgos individuales: 0.0070
Riesgo real del portafolio combinado: 0.0036


El riesgo real del portafolio (≈0.36%) es **casi la mitad** de lo que esperarías si solo promediaras el riesgo de cada activo por separado (≈0.70%). Ningún activo individual cambió de riesgo — el portafolio completo sí.

**Explica con tus palabras**, en una celda de texto, por qué combinar activos distintos puede reducir el riesgo total sin que ninguno de ellos se vuelva menos riesgoso por su cuenta. (Pista: fíjate en qué días TSLA pierde y qué hacen AAPL o MSFT esos mismos días.)

<details>
<summary>💡 Ver una forma de responder</summary>

Porque los activos no suben y bajan exactamente el mismo día ni en la misma dirección. Cuando uno pierde, otro puede sostenerse o ganar, y eso amortigua el resultado combinado — es el efecto de la **diversificación**.

Dicho de otra forma: el riesgo de un portafolio no depende solo de cuánto se mueve cada activo por separado, sino de qué tan parecido se mueven entre sí — su **correlación**. Cuando dos activos tienden a moverse en direcciones opuestas (uno sube cuando el otro baja), combinarlos reduce el riesgo total mucho más que si simplemente promediaras sus riesgos individuales. Es la razón por la que un portafolio bien armado casi siempre mezcla activos que no reaccionan igual a las mismas condiciones del mercado.
</details>

**✏️ Pruébalo tú mismo** — cambia los pesos abajo (deben sumar 1.0) y observa qué le pasa al riesgo del portafolio.

In [22]:
mis_pesos = np.array([0.25, 0.25, 0.25, 0.25])  # cambia estos números — deben sumar 1.0

mi_portafolio = AAPL*mis_pesos[0] + MSFT*mis_pesos[1] + AMZN*mis_pesos[2] + TSLA*mis_pesos[3]
print(f"Rendimiento: {mi_portafolio.mean():.4f}  |  Riesgo: {mi_portafolio.std():.4f}")

Rendimiento: 0.0032  |  Riesgo: 0.0070


---
## 🎯 07 · Reto individual

Retoma el portafolio que te asignaron en el reto de S11-S12.

In [ ]:
portafolios = {
    0: {"AAPL": {"cantidad": 10, "precio_compra": 180}, "MSFT": {"cantidad": 8, "precio_compra": 400}, "AMZN": {"cantidad": 12, "precio_compra": 170}, "GOOGL": {"cantidad": 6, "precio_compra": 150}},
    1: {"TSLA": {"cantidad": 15, "precio_compra": 220}, "NVDA": {"cantidad": 20, "precio_compra": 120}, "META": {"cantidad": 10, "precio_compra": 480}, "NFLX": {"cantidad": 5,  "precio_compra": 600}},
    2: {"JPM":  {"cantidad": 25, "precio_compra": 150}, "KO":   {"cantidad": 40, "precio_compra": 60},  "DIS":  {"cantidad": 20, "precio_compra": 95},  "INTC": {"cantidad": 50, "precio_compra": 35}},
    3: {"V":    {"cantidad": 12, "precio_compra": 260}, "PG":   {"cantidad": 18, "precio_compra": 150}, "XOM":  {"cantidad": 30, "precio_compra": 105}, "PFE":  {"cantidad": 60, "precio_compra": 30}},
    4: {"AMD":  {"cantidad": 22, "precio_compra": 140}, "CSCO": {"cantidad": 35, "precio_compra": 50},  "IBM":  {"cantidad": 10, "precio_compra": 190}, "ORCL": {"cantidad": 15, "precio_compra": 130}},
    5: {"BA":   {"cantidad": 8,  "precio_compra": 210}, "CAT":  {"cantidad": 10, "precio_compra": 320}, "MMM":  {"cantidad": 20, "precio_compra": 100}, "GE":   {"cantidad": 25, "precio_compra": 160}},
    6: {"WMT":  {"cantidad": 15, "precio_compra": 65},  "HD":   {"cantidad": 8,  "precio_compra": 340}, "NKE":  {"cantidad": 30, "precio_compra": 90},  "MCD":  {"cantidad": 10, "precio_compra": 280}},
    7: {"T":    {"cantidad": 60, "precio_compra": 18},  "VZ":   {"cantidad": 40, "precio_compra": 40},  "CVX":  {"cantidad": 12, "precio_compra": 150}, "JNJ":  {"cantidad": 15, "precio_compra": 160}},
}
precios_actuales_todos = {
    0: {"AAPL": 195, "MSFT": 385, "AMZN": 185, "GOOGL": 165},
    1: {"TSLA": 250, "NVDA": 135, "META": 460, "NFLX": 640},
    2: {"JPM": 165, "KO": 58, "DIS": 100, "INTC": 28},
    3: {"V": 280, "PG": 155, "XOM": 115, "PFE": 26},
    4: {"AMD": 160, "CSCO": 48, "IBM": 210, "ORCL": 145},
    5: {"BA": 195, "CAT": 350, "MMM": 95, "GE": 180},
    6: {"WMT": 70, "HD": 360, "NKE": 78, "MCD": 295},
    7: {"T": 20, "VZ": 38, "CVX": 165, "JNJ": 172},
}

# 👉 Reemplaza con tu código de estudiante (solo números)
mi_codigo_estudiante = 000000

grupo = mi_codigo_estudiante % 8
mi_portafolio = portafolios[grupo]
mis_precios = precios_actuales_todos[grupo]

print(f"Tu portafolio asignado (grupo {grupo}):")
print(mi_portafolio)

### Tu tarea

Con `mi_portafolio` y `mis_precios`, usando **solo arrays de NumPy**:

1. Calcula `valor_invertido`, `valor_actual`, `ganancia` y `rendimiento` (%) de tus 4 activos.
2. Calcula el promedio, el mejor, el peor y la desviación estándar del rendimiento.
3. **Escribe una recomendación de una sola frase** para tu portafolio, dirigida a un inversionista, que considere tanto el rendimiento como el riesgo — no solo uno de los dos. Debe quedar claro, leyendo tu frase, en qué números te basaste.

In [42]:
# ✏️ Tu código va aquí — usa las secciones 04-06 como referencia, no hay plantilla.
import numpy as np

print("NumPy", np.__version__, "listo para usar.")

precios = np.array([331,202,557,143])

print(precios)
print("shape:", precios.shape)
print("dtype:", precios.dtype)

precios_5_dias = np.array([
    [327, 329, 330, 332, 331],   # AAPL
    [198, 200, 205, 203, 202],   # TSLA
])

print(precios_5_dias)
print("shape:", precios_5_dias.shape, "  # (2 activos, 5 días)")

cantidades = np.array([10, 20, 6, 15])   # AAPL, TSLA, GOOGL, ADBE
rendimientos = np.array([16.55, 6.32, 25.73, 16.26])  # % de ganancia

np.array([1, 2, 3, 4])

precios_actuales = np.array([331, 202, 557, 143])   # AAPL, TSLA, GOOGL, ADBE
precios_compra = np.array([284, 190, 443, 123])     # AAPL, TSLA, GOOGL, ADBE

valor_invertido = cantidades * precios_compra
valor_actual = cantidades * precios_actuales

print("Valor actual:", valor_actual)

ganancia = valor_actual - valor_invertido
en_perdida = ganancia < 0

print("Ganancia:", ganancia)
print("En pérdida:", en_perdida)

AAPL = np.array([0.012, 0.008, -0.004, 0.015, 0.006, -0.002, 0.010, 0.003])

print("Número de días:", len(AAPL))
print("Primer día:", AAPL[0])
print("Último día:", AAPL[-1])

promedio_aapl = AAPL.mean()
peor_dia = AAPL.min()

print("Promedio:", promedio_aapl)
print("Peor día:", peor_dia)

riesgo_aapl = AAPL.std()
print("Riesgo (std):", riesgo_aapl)

AAPL  = np.array([0.012, 0.008, -0.004, 0.015, 0.006, -0.002, 0.010, 0.003])
TSLA  = np.array([0.030, -0.025, 0.040, 0.015, -0.020, 0.035, -0.010, 0.020])
GOOGL = np.array([0.004, 0.003, 0.005, 0.002, 0.006, 0.004, 0.003, 0.005])
ADBE  = np.array([-0.010, -0.015, 0.002, -0.008, -0.012, 0.005, -0.020, -0.006])

activos = {"AAPL": AAPL, "TSLA": TSLA, "GOOGL": GOOGL, "ADBE": ADBE}

for nombre, serie in activos.items():
    promedio = serie.mean()
    riesgo = serie.std()
    print(f"{nombre}: rendimiento {promedio:.4f}  |  riesgo {riesgo:.4f}")

pesos = np.array([0.30, 0.25, 0.20, 0.25])  # AAPL, TSLA, GOOGL, ADBE — suman 1.0

rendimiento_portafolio = AAPL*pesos[0] + TSLA*pesos[1] + GOOGL*pesos[2] + ADBE*pesos[3]
print(rendimiento_portafolio)

rendimiento_promedio_portafolio = rendimiento_portafolio.mean()
riesgo_portafolio = rendimiento_portafolio.std()

print("Rendimiento promedio:", rendimiento_promedio_portafolio)
print("Riesgo:", riesgo_portafolio)

riesgos_individuales = np.array([AAPL.std(), TSLA.std(), GOOGL.std(), ADBE.std()])
riesgo_promedio_simple = (riesgos_individuales * pesos).sum()

riesgo_portafolio = rendimiento_portafolio.std()

print(f"Riesgo si solo promediaras los riesgos individuales: {riesgo_promedio_simple:.4f}")
print(f"Riesgo real del portafolio combinado: {riesgo_portafolio:.4f}")

mis_pesos = np.array([0.25, 0.25, 0.25, 0.25])  # AAPL, TSLA, GOOGL, ADBE — suman 1.0

mi_portafolio = AAPL*mis_pesos[0] + TSLA*mis_pesos[1] + GOOGL*mis_pesos[2] + ADBE*mis_pesos[3]
print(f"Rendimiento: {mi_portafolio.mean():.4f}  |  Riesgo: {mi_portafolio.std():.4f}")



NumPy 2.1.3 listo para usar.
[331 202 557 143]
shape: (4,)
dtype: int64
[[327 329 330 332 331]
 [198 200 205 203 202]]
shape: (2, 5)   # (2 activos, 5 días)
Valor actual: [3310 4040 3342 2145]
Ganancia: [470 240 684 300]
En pérdida: [False False False False]
Número de días: 8
Primer día: 0.012
Último día: 0.003
Promedio: 0.006
Peor día: -0.004
Riesgo (std): 0.006224949798994366
AAPL: rendimiento 0.0060  |  riesgo 0.0062
TSLA: rendimiento 0.0106  |  riesgo 0.0239
GOOGL: rendimiento 0.0040  |  riesgo 0.0012
ADBE: rendimiento -0.0080  |  riesgo 0.0078
[ 0.0094  -0.007    0.0103   0.00665 -0.005    0.0102  -0.0039   0.0054 ]
Rendimiento promedio: 0.00325625
Riesgo: 0.006857998865376109
Riesgo si solo promediaras los riesgos individuales: 0.0100
Riesgo real del portafolio combinado: 0.0069
Rendimiento: 0.0032  |  Riesgo: 0.0070


### Recomendacion
Con un rendimiento promedio diario de 0.32 % y una desviación estándar de apenas 0.0070, este portafolio ofrece una ganancia consistente sin exponer al inversionista a una volatilidad significativa, gracias a la diversificación entre TSLA (más volátil) y GOOGL (más estable).

### Exit ticket

«El rendimiento promedio de mi portafolio es **___** %. La desviación estándar es **___**. Mi recomendación para un inversionista es ___, porque ___.»

---

📚 **Fuentes** · [NumPy quickstart](https://numpy.org/doc/stable/user/quickstart.html) · [What is NumPy?](https://numpy.org/doc/stable/user/whatisnumpy.html) · ["La desviación estándar: cálculo e interpretación para invertir", inbestMe](https://www.inbestme.com/es/es/blog/desviacion-estandar-volatilidad-riesgo/) · ["¿Qué significa la desviación estándar respecto a los riesgos de inversión?", Cuida tu Dinero](https://www.cuidatudinero.com/13168060/que-significa-la-desviacion-estandar-respecto-a-los-riesgos-de-inversion)